### ISSUE TITLE
ENDPOINT QUERY() param optional

### ISSUE THREAD

**Issue by aurickislam** [oatpp/oatpp#411]
Hello

I have declared **queryParams["active"]** as **optional (info->queryParams["active"].required = false;)**. It was optional on swagger ui but still **required** on server side. I was getting 400 response if url don't contain "active" param. How can I make it **optional**? The following is my code.

``` cpp
    ENDPOINT_INFO(getUser)
    {
        // general
        info->summary = "Get one User by username";
        info->addResponse<Object<UserDto>>(Status::CODE_200, "application/json");
        info->addResponse<Object<ErrorResponseDto>>(Status::CODE_404, "application/json");
        // params specific
        info->pathParams["username"].description = "username/login";
        
        // info->queryParams.add<Boolean>("active").description = "Active or Inactive";
        info->queryParams["active"].description = "Active or Inactive";
        info->queryParams["active"].required = false;
    }
    ENDPOINT("GET", "api/v1/users/{username}", getUser, PATH(String, username), QUERY(Boolean, active, "active"))
    {
        OATPP_LOGD("active", "active = %d", *active);
        
        auto user = m_database->getUser(username);

        OATPP_LOGD("user", jsonUtils->toPrettyJSON(user));

        if (!user)
            return createErroeResponse(Status::CODE_404, "User not found");

        return createDtoResponse(Status::CODE_200, user);
    }
```

**Comment by lganzzzo** [comment]
Hey @aurickislam ,

To make a query parameter optional - define a default value for that parameter in the `QUERY` macro:

```cpp
ENDPOINT("GET", "api/v1/users/{username}", getUser, PATH(String, username), QUERY(Boolean, active, "active", true))
```

**Comment by aurickislam** [comment]
Thanks it worked. But I had to pass **defaultValue** as **String**.

As **defaultValue** type is **const oatpp::String&**. It is defined on "oatpp/web/protocol/http/incoming/Request.hpp" file.
``` cpp
oatpp::String getQueryParameter(const oatpp::data::share::StringKeyLabel& name, const oatpp::String& defaultValue) const;
```
The correct format is
``` cpp
ENDPOINT("GET", "api/v1/users/{username}", getUser, PATH(String, username), QUERY(Boolean, active, "active", "true"))
```
The **defaultValue** feature is not documented yet. Is it possible to add it on the documentation?
And is it possible to make the **defaultValue** type Generic?

Thanks

**Comment by lganzzzo** [comment]
Hey @aurickislam ,

Thanks for the notes. I'll schedule it for `1.3.0`.

**Comment by lganzzzo** [comment]
Default value type is fixed in `1.3.0`
